pkg/network/pod/annotations/generator.go

```diff
@@ -100,17 +103,55 @@ func (g Generator) GenerateFromSource(vmi *v1.VirtualMachineInstance, sourcePod
 }
 
 // GenerateFromActivePod generates additional pod annotations, bases on information that exists on a live virt-launcher pod
-func (g Generator) GenerateFromActivePod(vmi *v1.VirtualMachineInstance, pod *k8Scorev1.Pod) map[string]string {
+func (g Generator) GenerateFromActivePod(vmi *v1.VirtualMachineInstance, pod *k8scorev1.Pod) map[string]string {
 	annotations := map[string]string{}
 
 	if deviceInfoAnnotation := g.generateDeviceInfoAnnotation(vmi, pod); deviceInfoAnnotation != "" {
 		annotations[downwardapi.NetworkInfoAnnot] = deviceInfoAnnotation
 	}
 
+	if updatedMultusAnnotation := g.generateMultusAnnotation(vmi, pod); updatedMultusAnnotation != nil {
+		annotations[networkv1.NetworkAttachmentAnnot] = *updatedMultusAnnotation
+	}
+
 	return annotations
 }
 
-func (g Generator) generateDeviceInfoAnnotation(vmi *v1.VirtualMachineInstance, pod *k8Scorev1.Pod) string {
+func (g Generator) generateMultusAnnotation(vmi *v1.VirtualMachineInstance, pod *k8scorev1.Pod) *string {
+	vmiSpecIfaces, vmiSpecNets, dynamicIfacesExist := vmispec.CalculateInterfacesAndNetworksForMultusAnnotationUpdate(vmi)
+	if !dynamicIfacesExist {
+		return nil
+	}
+
+	podIfaceNamesByNetworkName := namescheme.CreateFromNetworkStatuses(vmiSpecNets, multus.NetworkStatusesFromPod(pod))
+	updatedMultusAnnotation, err := multus.GenerateCNIAnnotationFromNameScheme(
+		vmi.Namespace,
+		vmiSpecIfaces,
+		vmiSpecNets,
+		podIfaceNamesByNetworkName,
+		g.clusterConfig,
+	)
+	if err != nil {
```

Comment:
I don't follow, could you please elaborate with an example?

Reply:
Since `updatedMultusAnnotation` is also used after the `if` statement, this is not feasible since it will limits its scope to within the `if` statement.